Source/Core/Core/MemoryWatcher.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <iostream>
+
+#include "Common/FileUtil.h"
+#include "Common/Thread.h"
+#include "Core/MemoryWatcher.h"
+#include "Core/HW/Memmap.h"
+
+// We don't want to kill the cpu, so sleep for this long after polling.
+static const int SLEEP_DURATION = 10; // ms
+
+MemoryWatcher::MemoryWatcher()
+{
+	if (!LoadAddresses(File::GetUserPath(F_MEMORYWATCHERLOCATIONS_IDX)))
+		return;
+	if (!OpenSocket(File::GetUserPath(F_MEMORYWATCHERSOCKET_IDX)))
+		return;
+	m_running = true;
+	m_watcher_thread = std::thread(&MemoryWatcher::WatcherThread, this);
+}
+
+MemoryWatcher::~MemoryWatcher()
+{
+	if (!m_running)
+		return;
+
+	m_running = false;
+	m_watcher_thread.join();
+	close(m_fd);
+}
+
+bool MemoryWatcher::LoadAddresses(const std::string& path)
+{
```

Comment:
Shouldn't need to, since I'm creating a new MemoryWatcher instance for each core init and deleting it on core shutdown.